Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,173 @@ This document includes four main sections:
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will create a database of Monty Python movies
+using basic :mod:`!sqlite3` functionality.
+It assumes a fundamental understanding of database concepts,
+including `cursors`_ and `transactions`_.
+
+First, we need to create a new database and open
+a database connection to allow :mod:`!sqlite3` to work with it.
+Call :func:`sqlite3.connect` to to create a connection to
+the database :file:`tutorial.db` in the current working directory,
+implicitly creating it if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the on-disk database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+we will need to use a database cursor.
+Call :meth:`con.cursor() <Connection.cursor>` to create the :class:`Cursor`::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
-
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+Now that we've got a database connection and a cursor,
+we can create a database table ``movie`` with columns for title,
+release year, and review score.
+For simplicity, we can just use column names in the table declaration --
+thanks to the `flexible typing`_ feature of SQLite,
+specifying the data types is optional.
+Execute the ``CREATE TABLE`` statement
+by calling :meth:`cur.execute(...) <Cursor.execute>`::
+
+   cur.execute("CREATE TABLE movie(title, year, score)")
+
+.. Ideally, we'd use sqlite_schema instead of sqlite_master below,
+   but SQLite versions older than 3.33.0 do not recognise that variant.
+
+We can verify that the new table has been created by querying
+the ``sqlite_master`` table built-in to SQLite,
+which should now contain an entry for the ``movie`` table definition
+(see `The Schema Table`_ for details).
+Execute that query by calling :meth:`cur.execute(...) <Cursor.execute>`,
+assign the result to ``res``,
+and call :meth:`res.fetchone() <Cursor.fetchone>` to fetch the resulting row::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master")
+   >>> res.fetchone()
+   ('movie',)
+
+The query shows that the table has been created,
+as it returns a :class:`tuple` containing the table's name.
+If we query ``sqlite_master`` for a non-existent table ``spam``,
+:meth:`!res.fetchone()` will return ``None``::
+
+   >>> res = cur.execute("SELECT name FROM sqlite_master WHERE name='spam'")
+   >>> res.fetchone()
+   >>>
```

Comment:
```suggestion
   >>> print(res.fetchone())
   None
```

Be more clear and explicit here about what is going on—I could see how a beginner might be confused by nothing being printed and just returning the prompt.

Reply:
This feels a bit unnecessary and distracting, even if more explicit.  Someone learning `sqlite3` should already be somewhat familiar with how the interactive interpreter works.  Even if they don't understand that the method returns `None`  and the interactive interpreter ignores it, the point of the example is still clear: there are no results.

Reply:
I could see the argument if it made the example much more verbose, but in this case its just adding a simple `print()` and replacing a prompt with the output `None`, so I don't really understand how being clearer and more explicit about the intent here is "distracting". In fact, I'd say I'm a fairly experienced Python user, and this behavior still confuses me (if momentarily) once and a while.

Using an explicit `print()` where printed output is required makes the intent and the result clearer and more unambiguous, per the Diataxis tutorial principle of [Ensure the user sees results immediately](https://diataxis.fr/tutorials/#ensure-the-user-sees-results-immediately):

> As far as possible, the effect of every action should be clear to them as soon as possible. The relation of cause and effect should be evident. Finally, each result should be something that the user can see as meaningful. **Every step the learner follows should produce a comprehensible result, however small.**

Furthermore, an explicit `print()` works regardless of how the user is running their code (not just directly line by line at the interactive `python` interpreter prompt), in keeping with the Diataxis tutorial principles of [Ensure that the tutorial works reliably](https://diataxis.fr/tutorials/#ensure-that-the-tutorial-works-reliably):

> The learner needs to see that when they follow your directions, they will attain the results you promise.

and [Make your tutorial repeatable](https://diataxis.fr/tutorials/#make-your-tutorial-repeatable):

> Unless you’re very lucky, the users of your tutorial will have different levels of skill and understanding. They might also be using different tools and operating systems and you can’t rely on them having the same resources or environment.

It is also consistent with all the other examples that produce output, save one (the other `cur.fetchone`, which would be good to also update).

Finally, the [Devguide explicitly proscribes using an empty prompt as the last line](https://devguide.python.org/documentation/markup/#showing-code-examples):

>After the last line of input or output presented, there should not be an “unused” primary prompt; this is an example of what not to do:
>
> ```pycon
> >>> 1 + 1
> 2
> >>>
> ```

Reply:
I'd use a simple test instead of a `print`:

```suggestion
   >>> res.fetchone() == None
   True
```

An `assert` would also do.

Reply:
That also works...though it seems a little inconsistent with what the other instances do?

Reply:
Also, of course:

```suggestion
   >>> res.fetchone() is None
   True
```

Reply:
Doh

Reply:
Keeping me on my toes :laughing: